workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Improbable.Gdk.Core;
+using Improbable.Gdk.Mobile;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Playground
+{
+    public class ConnectionScreenController : MonoBehaviour
+    {
+        private const string CachedIp = "cachedIp";
+
+        [SerializeField] private GameObject connectionPanel;
+        [SerializeField] private InputField ipAddressInput;
+        [SerializeField] private Button connectButton;
+        [SerializeField] private Text errorMessage;
+        [SerializeField] private GameObject clientWorkerPrefab;
+
+        private GameObject worker;
+
+        private string IpAddress => ipAddressInput != null ? ipAddressInput.text : null;
```

Comment:
`ipAddressInput?.text`

Reply:
^ This doesn't work because of Unity complaining about their underlying null checks or something like that.
```'?.' on a type deriving from 'UnityEngine.Object' bypasses the lifetime check on the underlying Unity engine object```

Reply:
Ah, yes :(